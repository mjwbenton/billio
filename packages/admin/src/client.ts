import { ApolloClient, InMemoryCache } from "@apollo/client";

const CLIENT = new ApolloClient({
  uri: "http://localhost:4000/",
  cache: new InMemoryCache(),
  defaultOptions: {
    query: {
      fetchPolicy: "network-only",
    },
  },
});

export default CLIENT;
