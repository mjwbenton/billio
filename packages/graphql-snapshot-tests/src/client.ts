import { ApolloClient, InMemoryCache, HttpLink } from "@apollo/client";
import fetch from "cross-fetch";

const ENDPOINT = process.env.BILLIO_ENDPOINT;

if (!ENDPOINT) {
  throw new Error("Please configure BILLIO_ENDPOINT for tests");
}

const CLIENT = new ApolloClient({
  cache: new InMemoryCache(),
  link: new HttpLink({
    uri: ENDPOINT,
    fetch,
  }),
});

export default CLIENT;
