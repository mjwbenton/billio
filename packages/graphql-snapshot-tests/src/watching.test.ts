import client from "./client";
import { gql } from "@apollo/client";

let MOVIE_ID: string = "";
let SERIES_ID: string = "";
let SEASON_ID: string = "";

jest.setTimeout(20_000);

test("import data (setup)", async () => {
  const { data } = await client.mutate({
    mutation: gql`
      mutation Test_ImportData {
        importExternalMovie(externalId: "tmdb:153", shelfId: Watched) {
          id
        }
        importExternalTvSeason(
          externalId: "tmdbSeason:87917:1"
          shelfId: FinishedSeason
        ) {
          id
          series {
            id
          }
        }
      }
    `,
  });
  MOVIE_ID = data.importExternalMovie.id;
  SEASON_ID = data.importExternalTvSeason.id;
  SERIES_ID = data.importExternalTvSeason.series.id;
});

test("watching returns both movies and Tv", async () => {
  const { data } = await client.query({
    query: gql`
      query Test_Watching {
        watching(first: 10) {
          total
          items {
            __typename
            ... on Movie {
              title
              releaseYear
            }
            ... on TvSeries {
              title
              releaseYear
              seasons {
                title
                seasonNumber
              }
            }
          }
        }
      }
    `,
  });
  expect(data).toMatchSnapshot();
});

test("can delete data (cleanup)", async () => {
  await client.mutate({
    mutation: gql`
      mutation Test_DeleteMovie($id: ID!) {
        deleteMovie(id: $id) {
          id
        }
      }
    `,
    variables: {
      id: MOVIE_ID,
    },
  });
  await client.mutate({
    mutation: gql`
      mutation Test_DeleteSeason($id: ID!) {
        deleteTvSeason(id: $id) {
          id
        }
      }
    `,
    variables: {
      id: SEASON_ID,
    },
  });
  await client.mutate({
    mutation: gql`
      mutation Test_DeleteSeries($id: ID!) {
        deleteTvSeries(id: $id) {
          id
        }
      }
    `,
    variables: {
      id: SERIES_ID,
    },
  });
});
