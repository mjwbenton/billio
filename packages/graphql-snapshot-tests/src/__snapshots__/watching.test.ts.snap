// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watching returns both movies and Tv 1`] = `
Object {
  "watching": Object {
    "__typename": "WatchingPage",
    "items": Array [
      Object {
        "__typename": "TvSeries",
        "releaseYear": "2019",
        "seasons": Array [
          Object {
            "__typename": "TvSeason",
            "rewatch": false,
            "seasonNumber": 1,
            "title": "For All Mankind: Season 1",
          },
        ],
        "title": "For All Mankind",
      },
      Object {
        "__typename": "Movie",
        "releaseYear": "2003",
        "title": "Lost in Translation",
      },
    ],
    "total": 2,
  },
}
`;
